console/console-init/ui/src/modules/msg-and-iot/components/ProjectList/ProjectList.tsx

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import {
+  Table,
+  TableVariant,
+  TableHeader,
+  TableBody,
+  SortByDirection,
+  IRowData,
+  sortable,
+  TableProps
+} from "@patternfly/react-table";
+import { Link } from "react-router-dom";
+import { FormatDistance } from "use-patternfly";
+import { StyleSheet, css } from "@patternfly/react-styles";
+import {
+  StatusTypes,
+  ProjectStatus,
+  ProjectError,
+  ProjectEntities,
+  ProjectTypePlan,
+  ProjectTypes
+} from "modules/msg-and-iot/utils";
+
+export const StyleForTable = StyleSheet.create({
+  scroll_overflow: {
+    overflowY: "auto",
+    paddingBottom: 100
+  }
+});
+
+export interface IProject {
+  projectType: ProjectTypes;
+  name?: string;
+  displayName?: string;
+  namespace?: string;
+  plan?: string;
+  status?: StatusTypes;
+  creationTimestamp?: string;
+  errorMessageRate?: number;
+  addressCount?: number;
+  connectionCount?: number;
+  deviceCount?: number;
+  activeCount?: number;
+  selected?: boolean;
+  errorMessages?: Array<string>;
+}
+
+export interface IProjectListProps extends Pick<TableProps, "sortBy"> {
+  onSort?: (_event: any, index: number, direction: SortByDirection) => void;
+  projects: Array<IProject>;
```

Comment:
can we please try to use IProject[] instead of Array<IProject>.  IProject[] is more readable.